tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun.go

```diff
@@ -827,20 +829,29 @@ func computeIterations(run *v1alpha1.Run, tls *pipelineloopv1alpha1.PipelineLoop
 				iterationElements = append(iterationElements, v)
 			}
 		} else {
-			var strings []string
+			var stringArr []string
 			var ints []int
 			var dictsString []map[string]string
 			var dictsInt []map[string]int
-			errString := json.Unmarshal([]byte(iterationParamStr), &strings)
+			errString := json.Unmarshal([]byte(iterationParamStr), &stringArr)
 			errInt := json.Unmarshal([]byte(iterationParamStr), &ints)
 			errDictString := json.Unmarshal([]byte(iterationParamStr), &dictsString)
 			errDictInt := json.Unmarshal([]byte(iterationParamStr), &dictsInt)
 			if errString != nil && errInt != nil && errDictString != nil && errDictInt != nil {
-				return 0, iterationElements, fmt.Errorf("The value of the iterate parameter %q can not transfer to array", tls.IterateParam)
+				//try the default separator comma (,) in last
+				if strings.Contains(iterationParamStr, defaultIterationParamStrSeparator) {
```

Comment:
For this test case, it supposes to get to fail this condition and return the error message in the else block. However, it went into this block so the unit test failed by not seeing the expected failure.
https://github.com/kubeflow/kfp-tekton/blob/570c7de491655b69fb7237fdd4bb38ce7fd399d4/tekton-catalog/pipeline-loops/pkg/reconciler/pipelinelooprun/pipelinelooprun_test.go#L1025-L1045

Reply:
Thanks, it's my fault I didn't look into this, let me check.